pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -86,12 +85,123 @@ func (r *ReconcileStorageCluster) newOBCStorageClass(initData *ocsv1.StorageClus
 			"objectStoreName":      fmt.Sprintf("%s-object-store", initData.Namespace),
 			"objectStoreNamespace": initData.Namespace,
 			"region":               "us-east-1", // TODO: how to get the current region
->>>>>>> Adding OBC StorageClass in converged mode
 		},
 	}
 	return retSC
 }
 
+// ensureExternalClusterConfig ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalClusterConfig(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalClusterDetailsSecret,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ocsv1.ExternalResource
+	err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	// storageClasses stores all the StorageClasses coming from the JSON
+	var storageClasses map[string]map[string]string = make(map[string]map[string]string)
```

Comment:
Don't have a map for the data, just do `var storageClasses []ExternalResource` and then `storageClasses = append(storageClasses, d)`.